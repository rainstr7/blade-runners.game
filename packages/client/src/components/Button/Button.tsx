import { FC } from 'react'

export const Button: FC = () => {
  return (
    <button>Click me</button>
  )
}
