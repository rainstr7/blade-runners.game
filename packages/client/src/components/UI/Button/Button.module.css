.Button {
    background: #FFFFFF;
    border: 5px solid #000000;
    box-shadow: 0 4px 5px 4px rgba(0, 0, 0, 0.3);
    border-radius: 5px;
    padding: 29px 89px;
    font-size: 36px;
    line-height: 41px;
}
