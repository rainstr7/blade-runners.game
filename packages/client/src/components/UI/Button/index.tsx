import cn from './style.module.scss'
import { ButtonHTMLAttributes, FC } from 'react'

interface Props {
  size?: 'small' | 'medium' | 'large'
}

const Button: FC<ButtonHTMLAttributes<HTMLButtonElement> & Props> = ({
  children,
  size = 'medium',
  ...props
}) => {
  const buttonSize = cn[size]
  return (
    <button className={`${cn.Button} ${buttonSize}`} {...props}>
      {children}
    </button>
  )
}

export default Button
