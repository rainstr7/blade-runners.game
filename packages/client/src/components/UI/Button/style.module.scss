@import 'src/styles/vars';

.Button {
  font-family: $font-family;
  font-size: 1.8rem;
  font-weight: 800;
  padding: 1.5rem 6.5rem;
  border: $border;
  box-shadow: $box-shadow;
  background-color: $background-lite-color;
  border-radius: $border-radius;
}
