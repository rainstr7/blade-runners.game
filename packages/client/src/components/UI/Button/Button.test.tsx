import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import '@testing-library/jest-dom'

import Button from '.'

const user = userEvent.setup()

describe('Button', () => {
  test('Дэфолтный рендеринг', () => {
    render(<Button>Click me</Button>)

    expect(screen.getByRole('button')).toMatchSnapshot('default-button')
  })
  test('Большой размер', () => {
    render(<Button size="large">Click me</Button>)

    expect(screen.getByRole('button')).toMatchSnapshot('large-button')
  })

  test('Маленький размер', () => {
    render(
      <Button data-testid="small-button" size="small">
        Click me
      </Button>
    )

    expect(screen.getByTestId('small-button')).toMatchSnapshot('small-button')
  })
  test('Клик по кнопке', async () => {
    const onClickMock = jest.fn()
    render(<Button onClick={onClickMock}>Click me</Button>)
    const button = screen.getByText('Click me')
    await user.click(button)
    expect(onClickMock).toHaveBeenCalledTimes(1)
  })

  test('Текст на кнопке', () => {
    render(<Button data-testid="text-button">Text button</Button>)
    const button = screen.getByTestId('text-button')
    expect(button).toHaveTextContent('Text button')
  })
})
