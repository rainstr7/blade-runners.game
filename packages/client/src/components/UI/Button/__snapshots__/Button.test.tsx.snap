// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly large: large-button 1`] = `
<button
  class="undefined undefined"
>
  Click me
</button>
`;

exports[`Button renders correctly: default-button 1`] = `
<button
  class="undefined undefined"
>
  Click me
</button>
`;

exports[`Button size: small-button 1`] = `
<button
  class="undefined function small() { [native code] }"
  data-testid="small-button"
>
  Click me
</button>
`;
